src/coprocessor/endpoint.rs

```diff
@@ -59,24 +60,50 @@ const OUTDATED_ERROR_MSG: &'static str = "request outdated.";
 
 const ENDPOINT_IS_BUSY: &'static str = "endpoint is busy";
 
-pub struct Host {
+pub struct Host<R: CopSender + 'static> {
     engine: Box<Engine>,
     sched: Scheduler<Task>,
     reqs: HashMap<u64, Vec<RequestTask>>,
     last_req_id: u64,
-    pool: ThreadPool<CopContext>,
-    low_priority_pool: ThreadPool<CopContext>,
-    high_priority_pool: ThreadPool<CopContext>,
+    pool: ThreadPool<CopContext<R>>,
+    low_priority_pool: ThreadPool<CopContext<R>>,
+    high_priority_pool: ThreadPool<CopContext<R>>,
     max_running_task_count: usize,
 }
 
+pub type CopRequestStatistics = HashMap<u64, FlowStatistics>;
+
+pub trait CopSender: Send + Clone {
+    fn send(&self, CopRequestStatistics) -> RaftStoreResult<()>;
```

Comment:
Use your own Cop Result, don't use RaftStoreResult. 
IMO, you should only know RaftStore in server mod

Reply:
umm, CopResult also needs use `raftstore` mod, for implement `From` from the RaftStoreResult. I'm going to extract the PD work in another pr.

Reply:
You don't need to support From, you can use box_try or `if let Err(e) = ` + `format!` to return your own error.